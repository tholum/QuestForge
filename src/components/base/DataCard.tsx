import * as React from "react"
import { cva, type VariantProps } from "class-variance-authority"
import { MoreHorizontal, TrendingUp, TrendingDown, Target, Calendar } from "lucide-react"

import { cn } from "@/lib/utils"
import { Card, CardContent, CardDescription, CardHeader, CardTitle } from "@/components/ui/card"
import { Badge } from "@/components/ui/badge"
import { Progress } from "@/components/ui/progress"
import { Button } from "./Button"

const dataCardVariants = cva(
  "relative overflow-hidden transition-all duration-200 hover:shadow-md",
  {
    variants: {
      variant: {
        default: "border-border",
        primary: "border-primary/20 bg-primary/5",
        success: "border-green-200 bg-green-50 dark:border-green-800 dark:bg-green-950/30",
        warning: "border-yellow-200 bg-yellow-50 dark:border-yellow-800 dark:bg-yellow-950/30",
        danger: "border-red-200 bg-red-50 dark:border-red-800 dark:bg-red-950/30",
        info: "border-blue-200 bg-blue-50 dark:border-blue-800 dark:bg-blue-950/30",
      },
      size: {
        sm: "p-3",
        default: "p-4",
        lg: "p-6",
      },
      interactive: {
        true: "cursor-pointer hover:scale-[1.02] active:scale-[0.98]",
        false: "",
      },
    },
    defaultVariants: {
      variant: "default",
      size: "default",
      interactive: false,
    },
  }
)

export interface IDataCardProps extends
  React.HTMLAttributes<HTMLDivElement>,
  VariantProps<typeof dataCardVariants> {
  title: string
  description?: string
  value?: string | number
  subtitle?: string
  progress?: number
  progressLabel?: string
  badge?: {
    text: string
    variant?: "default" | "secondary" | "success" | "warning" | "destructive"
  }
  trend?: {
    direction: "up" | "down"
    value: string | number
    label?: string
  }
  metadata?: Array<{
    icon?: React.ReactNode
    label: string
    value: string
  }>
  actions?: Array<{
    label: string
    icon?: React.ReactNode
    onClick: () => void
    variant?: "default" | "secondary" | "ghost"
  }>
  showMenu?: boolean
  onMenuClick?: () => void
  loading?: boolean
  error?: string
}

const DataCard = React.forwardRef<HTMLDivElement, IDataCardProps>(
  ({
    className,
    variant,
    size,
    interactive,
    title,
    description,
    value,
    subtitle,
    progress,
    progressLabel,
    badge,
    trend,
    metadata,
    actions,
    showMenu = false,
    onMenuClick,
    loading = false,
    error,
    onClick,
    ...props
  }, ref) => {
    const TrendIcon = trend?.direction === "up" ? TrendingUp : TrendingDown
    const trendColor = trend?.direction === "up" ? "text-green-600" : "text-red-600"

    if (loading) {
      return (
        <Card ref={ref} className={cn(dataCardVariants({ variant, size, interactive: false }), className)}>
          <CardHeader className="animate-pulse">
            <div className="h-4 bg-muted rounded w-3/4"></div>
            <div className="h-3 bg-muted rounded w-1/2"></div>
          </CardHeader>
          <CardContent className="animate-pulse space-y-2">
            <div className="h-8 bg-muted rounded w-1/3"></div>
            <div className="h-2 bg-muted rounded"></div>
          </CardContent>
        </Card>
      )
    }

    if (error) {
      return (
        <Card ref={ref} className={cn(dataCardVariants({ variant: "danger", size, interactive: false }), className)}>
          <CardContent className="flex items-center justify-center p-6 text-red-600">
            <div className="text-center">
              <div className="text-sm font-medium">Error loading data</div>
              <div className="text-xs opacity-70 mt-1">{error}</div>
            </div>
          </CardContent>
        </Card>
      )
    }

    return (
      <Card
        ref={ref}
        className={cn(dataCardVariants({ variant, size, interactive }), className)}
        onClick={onClick}
        {...props}
      >
        <CardHeader className="flex flex-row items-start justify-between space-y-0 pb-2">
          <div className="flex-1 space-y-1">
            <div className="flex items-center gap-2">
              <CardTitle className="text-sm font-medium truncate">
                {title}
              </CardTitle>
              {badge && (
                <Badge variant={badge.variant} className="text-xs">
                  {badge.text}
                </Badge>
              )}
            </div>
            {description && (
              <CardDescription className="text-xs">
                {description}
              </CardDescription>
            )}
          </div>
          
          {showMenu && (
            <Button
              variant="ghost"
              size="icon-sm"
              onClick={(e) => {
                e.stopPropagation()
                onMenuClick?.()
              }}
              className="shrink-0"
            >
              <MoreHorizontal className="w-4 h-4" />
            </Button>
          )}
        </CardHeader>

        <CardContent className="space-y-3">
          {/* Main Value Display */}
          {value !== undefined && (
            <div className="flex items-center justify-between">
              <div>
                <div className="text-2xl font-bold tracking-tight">
                  {value}
                </div>
                {subtitle && (
                  <div className="text-xs text-muted-foreground">
                    {subtitle}
                  </div>
                )}
              </div>
              
              {trend && (
                <div className={cn("flex items-center gap-1 text-sm", trendColor)}>
                  <TrendIcon className="w-4 h-4" />
                  <span className="font-medium">{trend.value}</span>
                  {trend.label && (
                    <span className="text-xs opacity-70">{trend.label}</span>
                  )}
                </div>
              )}
            </div>
          )}

          {/* Progress Display */}
          {progress !== undefined && (
            <div className="space-y-2">
              <div className="flex items-center justify-between text-sm">
                <span className="text-muted-foreground">
                  {progressLabel || "Progress"}
                </span>
                <span className="font-medium">{progress}%</span>
              </div>
              <Progress value={progress} className="h-2" />
            </div>
          )}

          {/* Metadata */}
          {metadata && metadata.length > 0 && (
            <div className="grid grid-cols-1 gap-2 text-sm">
              {metadata.map((item, index) => (
                <div key={index} className="flex items-center gap-2 text-muted-foreground">
                  {item.icon}
                  <span className="text-xs">{item.label}:</span>
                  <span className="font-medium text-foreground ml-auto">
                    {item.value}
                  </span>
                </div>
              ))}
            </div>
          )}

          {/* Actions */}
          {actions && actions.length > 0 && (
            <div className="flex gap-2 pt-2">
              {actions.map((action, index) => (
                <Button
                  key={index}
                  variant={action.variant || "secondary"}
                  size="sm"
                  onClick={(e) => {
                    e.stopPropagation()
                    action.onClick()
                  }}
                  leftIcon={action.icon}
                  className="flex-1"
                >
                  {action.label}
                </Button>
              ))}
            </div>
          )}
        </CardContent>
      </Card>
    )
  }
)

DataCard.displayName = "DataCard"

export { DataCard, dataCardVariants }