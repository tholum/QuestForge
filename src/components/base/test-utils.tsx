import { render, RenderOptions } from '@testing-library/react'
import { ReactElement } from 'react'
import { vi } from 'vitest'

// Mock IntersectionObserver
global.IntersectionObserver = vi.fn(() => ({
  disconnect: vi.fn(),
  observe: vi.fn(),
  unobserve: vi.fn(),
}))

// Mock ResizeObserver
global.ResizeObserver = vi.fn(() => ({
  disconnect: vi.fn(),
  observe: vi.fn(),
  unobserve: vi.fn(),
}))

// Mock matchMedia
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: vi.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: vi.fn(), // deprecated
    removeListener: vi.fn(), // deprecated
    addEventListener: vi.fn(),
    removeEventListener: vi.fn(),
    dispatchEvent: vi.fn(),
  })),
})

// Mock navigator.vibrate for QuickAddButton tests
Object.defineProperty(navigator, 'vibrate', {
  writable: true,
  value: vi.fn(),
})

// Mock requestAnimationFrame
global.requestAnimationFrame = vi.fn(cb => setTimeout(cb, 16))
global.cancelAnimationFrame = vi.fn()

// Provider wrapper for components that might need context
const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <div className="bg-background text-foreground">
      {children}
    </div>
  )
}

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }

// Common test utilities
export const mockGoal = {
  id: 'test-goal-1',
  title: 'Test Goal',
  description: 'A test goal for components',
  isCompleted: false,
  isPaused: false,
  isCancelled: false,
  dueDate: new Date(Date.now() + 86400000), // Tomorrow
  startDate: new Date(),
  priority: 'medium' as const,
  category: 'fitness' as const,
  progress: 45,
  createdAt: new Date(),
  updatedAt: new Date(),
}

export const mockUser = {
  id: 'test-user-1',
  name: 'Test User',
  email: 'test@example.com',
  level: 5,
  xp: 1250,
  nextLevelXp: 1500,
  streak: 12,
  achievements: 8,
}

export const createMockGoal = (overrides?: Partial<typeof mockGoal>) => ({
  ...mockGoal,
  ...overrides,
})

// Accessibility test helpers
export const axeMatchers = {
  toHaveNoViolations: expect.toHaveNoViolations,
}

// Animation helpers for components with animations
export const waitForAnimation = (duration = 500) => 
  new Promise(resolve => setTimeout(resolve, duration))

// Component testing utilities
export const getByTextContent = (container: HTMLElement, text: string) => {
  return Array.from(container.querySelectorAll('*')).find(
    element => element.textContent === text
  )
}

export const queryByTextContent = (container: HTMLElement, text: string) => {
  return Array.from(container.querySelectorAll('*')).find(
    element => element.textContent === text
  ) || null
}

// Toast testing utilities (since toasts might be portaled)
export const findToastByText = (text: string) => {
  // Look in document body since toasts might be portaled
  const elements = Array.from(document.body.querySelectorAll('*'))
  return elements.find(el => el.textContent?.includes(text))
}

// Form testing utilities
export const fillAndSubmitForm = async (
  user: ReturnType<typeof import('@testing-library/user-event').default.setup>,
  fields: Record<string, string>
) => {
  for (const [label, value] of Object.entries(fields)) {
    const field = screen.getByLabelText(new RegExp(label, 'i'))
    await user.clear(field)
    await user.type(field, value)
  }
}

export * from '@testing-library/user-event'