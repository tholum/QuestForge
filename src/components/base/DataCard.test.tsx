import { render, screen, waitFor } from './test-utils'
import userEvent from '@testing-library/user-event'
import { describe, it, expect, vi } from 'vitest'
import { DataCard } from './DataCard'
import { Target, Clock } from 'lucide-react'

describe('DataCard', () => {
  it('renders basic card content', () => {
    render(
      <DataCard
        title="Test Card"
        description="Test Description"
        value="42"
        subtitle="items"
      />
    )

    expect(screen.getByText('Test Card')).toBeInTheDocument()
    expect(screen.getByText('Test Description')).toBeInTheDocument()
    expect(screen.getByText('42')).toBeInTheDocument()
    expect(screen.getByText('items')).toBeInTheDocument()
  })

  it('shows loading state', () => {
    render(<DataCard title="Loading Card" loading />)
    
    // Should show skeleton loading animation
    expect(screen.getByText('Loading Card').closest('[data-testid]') || 
           document.querySelector('.animate-pulse')).toBeInTheDocument()
  })

  it('shows error state', () => {
    render(<DataCard title="Error Card" error="Failed to load data" />)
    
    expect(screen.getByText('Error loading data')).toBeInTheDocument()
    expect(screen.getByText('Failed to load data')).toBeInTheDocument()
  })

  it('renders with progress bar', () => {
    render(
      <DataCard
        title="Progress Card"
        value="75"
        subtitle="percent complete"
        progress={75}
        progressLabel="Completion"
      />
    )

    expect(screen.getByText('Completion')).toBeInTheDocument()
    expect(screen.getByText('75%')).toBeInTheDocument()
    
    // Check for progress bar element
    expect(document.querySelector('[data-progress]') || 
           document.querySelector('[role="progressbar"]')).toBeInTheDocument()
  })

  it('renders with badge', () => {
    render(
      <DataCard
        title="Badge Card"
        value="5"
        badge={{ text: "Hot!", variant: "success" }}
      />
    )

    expect(screen.getByText('Hot!')).toBeInTheDocument()
  })

  it('renders with trend indicator', () => {
    render(
      <DataCard
        title="Trend Card"
        value="150"
        trend={{
          direction: "up",
          value: "+25%",
          label: "vs last week"
        }}
      />
    )

    expect(screen.getByText('+25%')).toBeInTheDocument()
    expect(screen.getByText('vs last week')).toBeInTheDocument()
  })

  it('renders metadata items', () => {
    render(
      <DataCard
        title="Metadata Card"
        value="42"
        metadata={[
          { icon: <Target data-testid="target-icon" />, label: "Goals", value: "5" },
          { icon: <Clock data-testid="clock-icon" />, label: "Time", value: "2h" }
        ]}
      />
    )

    expect(screen.getByTestId('target-icon')).toBeInTheDocument()
    expect(screen.getByTestId('clock-icon')).toBeInTheDocument()
    expect(screen.getByText('Goals:')).toBeInTheDocument()
    expect(screen.getByText('5')).toBeInTheDocument()
    expect(screen.getByText('Time:')).toBeInTheDocument()
    expect(screen.getByText('2h')).toBeInTheDocument()
  })

  it('renders action buttons', () => {
    const mockAction1 = vi.fn()
    const mockAction2 = vi.fn()

    render(
      <DataCard
        title="Actions Card"
        value="10"
        actions={[
          {
            label: "Edit",
            icon: <Target data-testid="edit-icon" />,
            onClick: mockAction1
          },
          {
            label: "Delete",
            onClick: mockAction2,
            variant: "secondary"
          }
        ]}
      />
    )

    expect(screen.getByRole('button', { name: /edit/i })).toBeInTheDocument()
    expect(screen.getByRole('button', { name: /delete/i })).toBeInTheDocument()
    expect(screen.getByTestId('edit-icon')).toBeInTheDocument()
  })

  it('handles action button clicks', async () => {
    const mockAction = vi.fn()
    const user = userEvent.setup()

    render(
      <DataCard
        title="Clickable Card"
        value="5"
        actions={[
          {
            label: "Click me",
            onClick: mockAction
          }
        ]}
      />
    )

    await user.click(screen.getByRole('button', { name: /click me/i }))
    expect(mockAction).toHaveBeenCalledTimes(1)
  })

  it('shows menu button when showMenu is true', () => {
    const mockMenuClick = vi.fn()

    render(
      <DataCard
        title="Menu Card"
        value="3"
        showMenu
        onMenuClick={mockMenuClick}
      />
    )

    expect(screen.getByRole('button', { name: /more/i }) || 
           screen.getByTestId('menu-button')).toBeInTheDocument()
  })

  it('handles menu button click', async () => {
    const mockMenuClick = vi.fn()
    const user = userEvent.setup()

    render(
      <DataCard
        title="Menu Card"
        value="3"
        showMenu
        onMenuClick={mockMenuClick}
      />
    )

    const menuButton = screen.getByRole('button', { name: /more/i }) || 
                      document.querySelector('[data-testid="menu-button"]') as HTMLElement
    
    if (menuButton) {
      await user.click(menuButton)
      expect(mockMenuClick).toHaveBeenCalledTimes(1)
    }
  })

  it('handles card click when interactive', async () => {
    const mockClick = vi.fn()
    const user = userEvent.setup()

    render(
      <DataCard
        title="Interactive Card"
        value="7"
        interactive
        onClick={mockClick}
      />
    )

    // Click on the card (but not on action buttons)
    const cardTitle = screen.getByText('Interactive Card')
    await user.click(cardTitle)
    
    expect(mockClick).toHaveBeenCalledTimes(1)
  })

  it('applies variant classes correctly', () => {
    const { rerender } = render(<DataCard title="Variant Card" variant="success" />)
    
    const card = screen.getByText('Variant Card').closest('div')
    expect(card).toHaveClass('border-green-200')
    
    rerender(<DataCard title="Variant Card" variant="danger" />)
    expect(card).toHaveClass('border-red-200')
  })

  it('applies size classes correctly', () => {
    const { rerender } = render(<DataCard title="Size Card" size="sm" />)
    
    const card = screen.getByText('Size Card').closest('div')
    expect(card).toHaveClass('p-3')
    
    rerender(<DataCard title="Size Card" size="lg" />)
    expect(card).toHaveClass('p-6')
  })

  it('applies interactive styling when interactive is true', () => {
    render(<DataCard title="Interactive Card" interactive />)
    
    const card = screen.getByText('Interactive Card').closest('div')
    expect(card).toHaveClass('cursor-pointer', 'hover:scale-[1.02]')
  })

  // Test event propagation for nested buttons
  it('prevents card click when action button is clicked', async () => {
    const mockCardClick = vi.fn()
    const mockActionClick = vi.fn()
    const user = userEvent.setup()

    render(
      <DataCard
        title="Nested Click Card"
        value="5"
        interactive
        onClick={mockCardClick}
        actions={[
          {
            label: "Action",
            onClick: mockActionClick
          }
        ]}
      />
    )

    // Click the action button - should not trigger card click
    await user.click(screen.getByRole('button', { name: /action/i }))
    
    expect(mockActionClick).toHaveBeenCalledTimes(1)
    expect(mockCardClick).not.toHaveBeenCalled()
  })

  // Accessibility tests
  it('has proper accessibility attributes', () => {
    render(
      <DataCard
        title="Accessible Card"
        description="This card is accessible"
        value="42"
      />
    )

    const card = screen.getByText('Accessible Card').closest('[role]')
    // Card should have appropriate role or be properly labeled
    expect(card).toBeInTheDocument()
  })

  it('supports keyboard navigation', async () => {
    const mockClick = vi.fn()
    const user = userEvent.setup()

    render(
      <DataCard
        title="Keyboard Card"
        interactive
        onClick={mockClick}
      />
    )

    const card = screen.getByText('Keyboard Card').closest('div')
    if (card) {
      card.focus()
      await user.keyboard('{Enter}')
      expect(mockClick).toHaveBeenCalledTimes(1)
    }
  })

  // Snapshot test
  it('matches snapshot for complex configuration', () => {
    const { container } = render(
      <DataCard
        title="Complex Card"
        description="Full-featured card"
        value="156"
        subtitle="total points"
        variant="primary"
        size="lg"
        progress={78}
        progressLabel="Progress"
        badge={{ text: "Featured", variant: "success" }}
        trend={{
          direction: "up",
          value: "+12%",
          label: "this week"
        }}
        metadata={[
          { label: "Category", value: "Fitness" },
          { label: "Streak", value: "5 days" }
        ]}
        actions={[
          {
            label: "View Details",
            onClick: () => {}
          }
        ]}
        showMenu
        interactive
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})