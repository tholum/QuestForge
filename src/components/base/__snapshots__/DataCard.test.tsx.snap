// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`DataCard > matches snapshot for complex configuration 1`] = `
<div
  class="bg-background text-foreground"
>
  <div
    class="text-card-foreground flex flex-col gap-6 rounded-xl border shadow-sm relative overflow-hidden transition-all duration-200 hover:shadow-md border-primary/20 bg-primary/5 p-6 cursor-pointer hover:scale-[1.02] active:scale-[0.98]"
    data-slot="card"
  >
    <div
      class="@container/card-header auto-rows-min grid-rows-[auto_auto] gap-1.5 px-6 has-data-[slot=card-action]:grid-cols-[1fr_auto] [.border-b]:pb-6 flex flex-row items-start justify-between space-y-0 pb-2"
      data-slot="card-header"
    >
      <div
        class="flex-1 space-y-1"
      >
        <div
          class="flex items-center gap-2"
        >
          <div
            class="text-sm font-medium truncate"
            data-slot="card-title"
          >
            Complex Card
          </div>
          <span
            class="inline-flex items-center justify-center rounded-md border px-2 py-0.5 font-medium w-fit whitespace-nowrap shrink-0 [&>svg]:size-3 gap-1 [&>svg]:pointer-events-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive transition-[color,box-shadow] overflow-hidden text-xs"
            data-slot="badge"
          >
            Featured
          </span>
        </div>
        <div
          class="text-muted-foreground text-xs"
          data-slot="card-description"
        >
          Full-featured card
        </div>
      </div>
      <button
        class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive focus:scale-[0.98] active:scale-[0.96] select-none hover:bg-accent hover:text-accent-foreground dark:hover:bg-accent/50 size-8 rounded-md shrink-0"
        data-loading="false"
        data-slot="button"
      >
        <span
          class="flex-1"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-ellipsis w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="12"
              cy="12"
              r="1"
            />
            <circle
              cx="19"
              cy="12"
              r="1"
            />
            <circle
              cx="5"
              cy="12"
              r="1"
            />
          </svg>
        </span>
      </button>
    </div>
    <div
      class="px-6 space-y-3"
      data-slot="card-content"
    >
      <div
        class="flex items-center justify-between"
      >
        <div>
          <div
            class="text-2xl font-bold tracking-tight"
          >
            156
          </div>
          <div
            class="text-xs text-muted-foreground"
          >
            total points
          </div>
        </div>
        <div
          class="flex items-center gap-1 text-sm text-green-600"
        >
          <svg
            aria-hidden="true"
            class="lucide lucide-trending-up w-4 h-4"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M16 7h6v6"
            />
            <path
              d="m22 7-8.5 8.5-5-5L2 17"
            />
          </svg>
          <span
            class="font-medium"
          >
            +12%
          </span>
          <span
            class="text-xs opacity-70"
          >
            this week
          </span>
        </div>
      </div>
      <div
        class="space-y-2"
      >
        <div
          class="flex items-center justify-between text-sm"
        >
          <span
            class="text-muted-foreground"
          >
            Progress
          </span>
          <span
            class="font-medium"
          >
            78
            %
          </span>
        </div>
        <div
          aria-valuemax="100"
          aria-valuemin="0"
          class="bg-primary/20 relative w-full overflow-hidden rounded-full h-2"
          data-max="100"
          data-slot="progress"
          data-state="indeterminate"
          role="progressbar"
        >
          <div
            class="bg-primary h-full w-full flex-1 transition-all"
            data-max="100"
            data-slot="progress-indicator"
            data-state="indeterminate"
            style="transform: translateX(-22%);"
          />
        </div>
      </div>
      <div
        class="grid grid-cols-1 gap-2 text-sm"
      >
        <div
          class="flex items-center gap-2 text-muted-foreground"
        >
          <span
            class="text-xs"
          >
            Category
            :
          </span>
          <span
            class="font-medium text-foreground ml-auto"
          >
            Fitness
          </span>
        </div>
        <div
          class="flex items-center gap-2 text-muted-foreground"
        >
          <span
            class="text-xs"
          >
            Streak
            :
          </span>
          <span
            class="font-medium text-foreground ml-auto"
          >
            5 days
          </span>
        </div>
      </div>
      <div
        class="flex gap-2 pt-2"
      >
        <button
          class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-all duration-200 disabled:pointer-events-none disabled:opacity-50 [&_svg]:pointer-events-none [&_svg:not([class*='size-'])]:size-4 [&_svg]:shrink-0 outline-none focus-visible:border-ring focus-visible:ring-ring/50 focus-visible:ring-[3px] aria-invalid:ring-destructive/20 dark:aria-invalid:ring-destructive/40 aria-invalid:border-destructive focus:scale-[0.98] active:scale-[0.96] select-none bg-secondary text-secondary-foreground shadow-xs hover:bg-secondary/80 hover:shadow-sm h-8 text-sm px-3 gap-1.5 has-[>svg]:px-2.5 rounded-md flex-1"
          data-loading="false"
          data-slot="button"
        >
          <span
            class="flex-1"
          >
            View Details
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
